### ISSUE TITLE
I am not able to run safari in Real Device

### ISSUE THREAD

**Issue by mashek347** [appium/appium#6908]
Launching Appium with command: '/Applications/Appium 3.app/Contents/Resources/node/bin/node' appium/build/lib/main.js --address "127.0.0.1" --debug-log-spacing --platform-version "9.3" --platform-name "iOS" --safari --browser-name "Safari" --udid "a132919a4b4c2c393ea15f7860fe8ae1e3d6f1e9" --show-ios-log --device-name "iPhone 6 Plus"

[Appium] Welcome to Appium v1.5.3

[Appium] Non-default server args:

[Appium]   address: '127.0.0.1'
[Appium]   safari: true

[Appium]   debugLogSpacing: true
[Appium]   platformName: 'iOS'
[Appium]   platformVersion: '9.3'
[Appium]   deviceName: 'iPhone 6 Plus'
[Appium]   browserName: 'Safari'
[Appium]   udid: 'a132919a4b4c2c393ea15f7860fe8ae1e3d6f1e9'
[Appium]   showIOSLog: true
[Appium] Deprecated server args:
[Appium]   --platform-name => --default-capabilities '{"platformName":"iOS"}'
[Appium]   --platform-version => --default-capabilities '{"platformVersion":"9.3"}'
[Appium]   --device-name => --default-capabilities '{"deviceName":"iPhone 6 Plus"}'

[Appium]   --browser-name => --default-capabilities '{"browserName":"Safari"}'
[Appium]   -U,--udid => --default-capabilities '{"udid":"a132919a4b4c2c393ea15f7860fe8ae1e3d6f1e9"}'
[Appium]   --show-ios-log => --default-capabilities '{"showIOSLog":true}'
[Appium] Default capabilities, which will be added to each request unless overridden by desired capabilities:

[Appium]   platformName: 'iOS'

[Appium]   platformVersion: '9.3'
[Appium]   deviceName: 'iPhone 6 Plus'
[Appium]   browserName: 'Safari'
[Appium]   udid: 'a132919a4b4c2c393ea15f7860fe8ae1e3d6f1e9'
[Appium]   showIOSLog: true

[Appium] Appium REST http interface listener started on 127.0.0.1:4723

[HTTP] --> GET /wd/hub/status {}

[MJSONWP] Calling AppiumDriver.getStatus() with args: []

[MJSONWP] Responding to client with driver.getStatus() result: {"build":{"version":"1.5.3"...

[HTTP] <-- GET /wd/hub/status 200 28 ms - 83 

[HTTP] --> POST /wd/hub/session {"desiredCapabilities":{"app":"io.appium.SafariLauncher","safariIgnoreFraudWarning":true,"platformVersion":"9.3","safariAllowPopups":false,"browserName":"safari","platformName":"iOS","udid":"a132919a4b4c2c393ea15f7860fe8ae1e3d6f1e9","deviceName":"iPhone 6"}}

[MJSONWP] Calling AppiumDriver.createSession() with args: [{"app":"io.appium.SafariLa...

[Appium] Creating new IosDriver session
[Appium] Capabilities:
[Appium]   app: 'io.appium.SafariLauncher'
[Appium]   safariIgnoreFraudWarning: true
[Appium]   platformVersion: '9.3'
[Appium]   safariAllowPopups: false
[Appium]   browserName: 'safari'
[Appium]   platformName: 'iOS'
[Appium]   udid: 'a132919a4b4c2c393ea15f7860fe8ae1e3d6f1e9'
[Appium]   deviceName: 'iPhone 6'
[Appium]   showIOSLog: true

[BaseDriver] Session created with session id: 9c3b8be8-899b-4ad9-9e14-c48b90ceabee
[debug] [iOS] Not auto-detecting udid.
[debug] [iOS] Creating iDevice object with udid a132919a4b4c2c393ea15f7860fe8ae1e3d6f1e9

[debug] [iOS] Checking for presence of SafariLauncher at '/Applications/Appium 3.app/Contents/Resources/node_modules/appium/node_modules/appium-ios-driver/build/SafariLauncher/SafariLauncher.app'

[debug] [iOS] SafariLauncher exists

[debug] [iOS] Removing any remaining instruments sockets
[debug] [iOS] Cleaned up instruments socket /var/folders/z3/tkbh3cns4fd4fyd6xsdgk1bh68_4ff/T/instruments_sock
[debug] [iOS] No language specified. Using default strings
[debug] [iOS] Strings file not found. Looking in 'en.lproj' directory
[iOS] Could not file localizable strings file 'Localizable.strings'!
[debug] [iOS] Creating instruments
[debug] [UIAuto] Preparing bootstrap code
[debug] [UIAuto] Dynamic bootstrap dir: /Users/502000533/Library/Application Support/appium/bootstrap
[debug] [UIAuto] Dynamic env: {"nodePath":"/Applications/Appium 3.app/Contents/Resources/node/bin/node","commandProxyClientPath":"/Applications/Appium 3.app/Contents/Resources/node_modules/appium/node_modules/appium-ios-driver/node_modules/appium-uiauto/build/lib/bin/command-proxy-client.js","instrumentsSock":"/var/folders/z3/tkbh3cns4fd4fyd6xsdgk1bh68_4ff/T/instruments_sock","interKeyDelay":null,"justLoopInfinitely":false,"autoAcceptAlerts":false,"autoDismissAlerts":false,"sendKeyStrategy":"grouped"}

[debug] [UIAuto] Dynamic bootstrap code: // This file is automatically generated. Do not manually modify!
...

[debug] [UIAuto] Dynamic bootstrap path: /Users/502000533/Library/Application Support/appium/bootstrap/bootstrap-7bf8a16bd619352a.js
[debug] [UIAuto] Reusing dynamic bootstrap: /Users/502000533/Library/Application Support/appium/bootstrap/bootstrap-7bf8a16bd619352a.js

[debug] [iOS] Running ios real device reset flow
[debug] [iOSLog] Attempting iOS device log capture via libimobiledevice idevicesyslog
[debug] [iOSLog] Found idevicesyslog: '/usr/local/bin/idevicesyslog'

[debug] [iOS] App is not installed. Will try to install.

[iOSLog] [IOS_SYSLOG_ROW] Sep 28 14:35:53 Alam-Mashekul-Genworth-Contractors-iPhone installd[50] <Notice>: 0x16e1bb000 -[MIClientConnection _doBackgroundInstallationForPath:withOptions:completion:]: Install of "/var/mobile/Media/PublicStaging/SafariLauncher.app" type Developer (LSInstallType = (null)) requested by mobile_installation_proxy (pid 286)

[iOSLog] [IOS_SYSLOG_ROW] Sep 28 14:35:53 Alam-Mashekul-Genworth-Contractors-iPhone installd[50] <Notice>: 0x16e1bb000 -[MIInstaller performInstallationWithError:]: Installing <MIInstallableBundle ID=com.bytearc.SafariLauncher; Version=2.0, ShortVersion=2.0>

[iOSLog] [IOS_SYSLOG_ROW] Sep 28 14:35:53 Alam-Mashekul-Genworth-Contractors-iPhone securityd[92] <Error>:  secTaskDiagnoseEntitlements MISSING keychain entitlements: no stored taskRef found
[iOSLog] [IOS_SYSLOG_ROW] Sep 28 14:35:53 Alam-Mashekul-Genworth-Contractors-iPhone securityd[92] <Error>:  secTaskDiagnoseEntitlements MISSING keychain entitlements: no stored taskRef found
[iOSLog] [IOS_SYSLOG_ROW] Sep 28 14:35:53 Alam-Mashekul-Genworth-Contractors-iPhone profiled[146] <Notice>: (Note ) MC: Provisioning profiles changed
[iOSLog] [IOS_SYSLOG_ROW] Sep 28 14:35:53 Alam-Mashekul-Genworth-Contractors-iPhone installd[50] <Error>:  SecTrustEvaluate  [leaf IssuerCommonName SubjectCommonName]
[iOSLog] [IOS_SYSLOG_ROW] Sep 28 14:35:53 Alam-Mashekul-Genworth-Contractors-iPhone securityd[92] <Error>:  secTaskDiagnoseEntitlements MISSING keychain entitlements: no stored taskRef found
[iOSLog] [IOS_SYSLOG_ROW] Sep 28 14:35:53 Alam-Mashekul-Genworth-Contractors-iPhone securityd[92] <Error>:  secTaskDiagnoseEntitlements MISSING keychain entitlements: no stored taskRef found
[iOSLog] [IOS_SYSLOG_ROW] Sep 28 14:35:53 Alam-Mashekul-Genworth-Contractors-iPhone installd[50] <Error>:  SecTrustEvaluate  [leaf IssuerCommonName SubjectCommonName]
[iOSLog] [IOS_SYSLOG_ROW] Sep 28 14:35:53 Alam-Mashekul-Genworth-Contractors-iPhone securityd[92] <Error>:  secTaskDiagnoseEntitlements MISSING keychain entitlements: no stored taskRef found
[iOSLog] [IOS_SYSLOG_ROW] Sep 28 14:35:53 Alam-Mashekul-Genworth-Contractors-iPhone securityd[92] <Error>:  secTaskDiagnoseEntitlements MISSING keychain entitlements: no stored taskRef found
[iOSLog] [IOS_SYSLOG_ROW] Sep 28 14:35:53 Alam-Mashekul-Genworth-Contractors-iPhone installd[50] <Error>:  SecTrustEvaluate  [leaf Revocation1]

[iOSLog] [IOS_SYSLOG_ROW] Sep 28 14:35:53 Alam-Mashekul-Genworth-Contractors-iPhone installd[50] <Error>: 0x16e1bb000 +[MICodeSigningVerifier _validateSignatureAndCopyInfoForURL:withOptions:error:]: 142: Failed to verify code signature of /private/var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.Wzsamd/extracted/SafariLauncher.app : 0xe8008018 (The identity used to sign the executable is no longer valid.)
[iOSLog] [IOS_SYSLOG_ROW] Sep 28 14:35:53 Alam-Mashekul-Genworth-Contractors-iPhone installd[50] <Error>: 0x16e1bb000 -[MIInstaller performInstallationWithError:]: Verification stage failed

[iOSLog] [IOS_SYSLOG_ROW] Sep 28 14:35:53 Alam-Mashekul-Genworth-Contractors-iPhone mobile_installation_proxy[286] <Error>: 0x16e087000 __MobileInstallationInstallForLaunchServices_block_invoke222: Returned error Error Domain=MIInstallerErrorDomain Code=13 "Failed to verify code signature of /private/var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.Wzsamd/extracted/SafariLauncher.app : 0xe8008018 (The identity used to sign the executable is no longer valid.)" UserInfo={LibMISErrorNumber=-402620392, LegacyErrorString=ApplicationVerificationFailed, SourceFileLine=142, FunctionName=+[MICodeSigningVerifier _validateSignatureAndCopyInfoForURL:withOptions:error:], NSLocalizedDescription=Failed to verify code signature of /private/var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.Wzsamd/extracted/SafariLauncher.app : 0xe8008018 (The identity used to sign the executable is no longer valid.)}
[iOSLog] [IOS_SYSLOG_ROW] Sep 28 14:35:53 Alam-Mashekul-Genworth-Contractors-iPhone mobile_installation_proxy[286] <Error>: 0x16e1bb000 handle_install: Installation failed: Error Domain=LaunchServicesError Code=0 "(null)" UserInfo={Error=ApplicationVerificationFailed, ErrorDetail=-402620392, ErrorDescription=Failed to verify code signature of /private/var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.Wzsamd/extracted/SafariLauncher.app : 0xe8008018 (The identity used to sign the executable is no longer valid.)}
[iOSLog] [IOS_SYSLOG_ROW] Sep 28 14:35:53 Alam-Mashekul-Genworth-Contractors-iPhone profiled[146] <Notice>: (Note ) MC: Updating MIS trust...

[MJSONWP] Encountered internal error running command: Error: Command failed: ideviceinstaller -u a132919a4b4c2c393ea15f7860fe8ae1e3d6f1e9 -i "/Applications/Appium 3.app/Contents/Resources/node_modules/appium/node_modules/appium-ios-driver/build/SafariLauncher/SafariLauncher.app"

```
at ChildProcess.exithandler (child_process.js:202:12)
at emitTwo (events.js:100:13)
at ChildProcess.emit (events.js:185:7)
at maybeClose (internal/child_process.js:850:16)
at Process.ChildProcess._handle.onexit (internal/child_process.js:215:5)
```

[HTTP] <-- POST /wd/hub/session 500 5108 ms - 359

**Comment by imurchie** [comment]
You need to update `ideviceinstaller`. It is not working.

**Comment by mashek347** [comment]
Still is not working. 

Please help me up?

**Comment by mashek347** [comment]
Last login: Thu Sep 29 10:30:42 on ttys004
GNWRICA2080432:~ 502000533$ appium &
[1] 3795
GNWRICA2080432:~ 502000533$ [Appium] Welcome to Appium v1.5.0-beta16 (REV 9378e46ffc2c17df4efc535744279ebfa28bcf30)
[Appium] Appium REST http interface listener started on 0.0.0.0:4723
[HTTP] --> POST /wd/hub/session 
[MJSONWP] Calling AppiumDriver.createSession() with args: [{"app":"io.appium.SafariLauncher","safariIgnoreFraudWarning":true,"platformVersion":"9.3","safariAllowPopups":false,"browserName":"safari","platfo...
[Appium] Creating new IosDriver session
[Appium] Capabilities:
[Appium]   app: 'io.appium.SafariLauncher'
[Appium]   safariIgnoreFraudWarning: true
[Appium]   platformVersion: '9.3'
[Appium]   safariAllowPopups: false
[Appium]   browserName: 'safari'
[Appium]   platformName: 'iOS'
[Appium]   udid: 'a132919a4b4c2c393ea15f7860fe8ae1e3d6f1e9'
[Appium]   deviceName: 'iPhone 6'
[BaseDriver] Session created with session id: 8f02a0b0-2cf2-4a35-8cd9-6d833829f77a
[debug] [iOS] Not auto-detecting udid.
[debug] [iOS] Checking for presence of SafariLauncher at '/usr/local/lib/node_modules/appium/node_modules/appium-ios-driver/build/SafariLauncher/SafariLauncher.app'
[debug] [iOS] SafariLauncher does not exist
[debug] [iOS] SafariLauncher not found, installing...
[iOS] Cleaning SafariLauncher
[debug] [iOS] Cleaning SafariLauncher for iphoneos9.3
[iOS] Error: Command 'xcodebuild -sdk iphoneos9.3 clean' errored out: Error: spawn xcodebuild ENOENT
    at exports._errnoException (util.js:870:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:178:32)
    at onErrorNT (internal/child_process.js:344:16)
    at nextTickCallbackWith2Args (node.js:442:9)
    at process._tickCallback (node.js:356:17)
    at ChildProcess.<anonymous> (lib/teen_process.js:31:14)
    at emitOne (events.js:77:13)
    at ChildProcess.emit (events.js:169:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
    at onErrorNT (internal/child_process.js:344:16)
    at nextTickCallbackWith2Args (node.js:442:9)
    at process._tickCallback (node.js:356:17)
 [Error: Command 'xcodebuild -sdk iphoneos9.3 clean' errored out: Error: spawn xcodebuild ENOENT
    at exports._errnoException (util.js:870:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:178:32)
    at onErrorNT (internal/child_process.js:344:16)
    at nextTickCallbackWith2Args (node.js:442:9)
    at process._tickCallback (node.js:356:17)]
[iOS] Error: Command 'xcodebuild -sdk iphoneos9.3 clean' errored out: Error: spawn xcodebuild ENOENT
    at exports._errnoException (util.js:870:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:178:32)
    at onErrorNT (internal/child_process.js:344:16)
    at nextTickCallbackWith2Args (node.js:442:9)
    at process._tickCallback (node.js:356:17)
    at ChildProcess.<anonymous> (lib/teen_process.js:31:14)
    at emitOne (events.js:77:13)
    at ChildProcess.emit (events.js:169:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
    at onErrorNT (internal/child_process.js:344:16)
    at nextTickCallbackWith2Args (node.js:442:9)
    at process._tickCallback (node.js:356:17)
 [Error: Command 'xcodebuild -sdk iphoneos9.3 clean' errored out: Error: spawn xcodebuild ENOENT
    at exports._errnoException (util.js:870:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:178:32)
    at onErrorNT (internal/child_process.js:344:16)
    at nextTickCallbackWith2Args (node.js:442:9)
    at process._tickCallback (node.js:356:17)]
[iOS] Bad app: '/usr/local/lib/node_modules/appium/node_modules/appium-ios-driver/build/SafariLauncher/SafariLauncher.app'. App paths need to be absolute, or relative to the appium server install dir, or a URL to compressed file, or a special app name.
[MJSONWP] Encountered internal error running command: Error: Bad app: /usr/local/lib/node_modules/appium/node_modules/appium-ios-driver/build/SafariLauncher/SafariLauncher.app. App paths need to be absolute, or relative to the appium server install dir, or a URL to compressed file, or a special app name.
    at IosDriver.configureApp$ (lib/driver.js:229:15)
    at tryCatch (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as throw] (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/usr/local/lib/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
    at run (/usr/local/lib/node_modules/appium/node_modules/core-js/library/modules/es6.promise.js:104:47)
    at /usr/local/lib/node_modules/appium/node_modules/core-js/library/modules/es6.promise.js:115:28
    at flush (/usr/local/lib/node_modules/appium/node_modules/core-js/library/modules/$.microtask.js:19:5)
    at nextTickCallbackWith0Args (node.js:420:9)
    at process._tickCallback (node.js:349:13)
[HTTP] <-- POST /wd/hub/session 500 188 ms - 381

**Comment by mashek347** [comment]
Here is the update appium log

Launching Appium with command: '/Applications/Appium 3.app/Contents/Resources/node/bin/node' appium/build/lib/main.js --address "127.0.0.1" --debug-log-spacing --platform-version "9.3" --platform-name "iOS" --safari --browser-name "Safari" --udid "334bde145520a3aed44d4399992a98a37461ce73" --full-reset --show-ios-log --default-device

[Appium] Welcome to Appium v1.5.3

[Appium] Non-default server args:

[Appium]   address: '127.0.0.1'

[Appium]   safari: true
[Appium]   defaultDevice: true
[Appium]   debugLogSpacing: true

[Appium]   platformName: 'iOS'
[Appium]   platformVersion: '9.3'
[Appium]   browserName: 'Safari'
[Appium]   udid: '334bde145520a3aed44d4399992a98a37461ce73'
[Appium]   fullReset: true
[Appium]   showIOSLog: true

[Appium] Deprecated server args:
[Appium]   --platform-name => --default-capabilities '{"platformName":"iOS"}'
[Appium]   --platform-version => --default-capabilities '{"platformVersion":"9.3"}'
[Appium]   --browser-name => --default-capabilities '{"browserName":"Safari"}'
[Appium]   -U,--udid => --default-capabilities '{"udid":"334bde145520a3aed44d4399992a98a37461ce73"}'
[Appium]   --full-reset => --default-capabilities '{"fullReset":true}'

[Appium]   --show-ios-log => --default-capabilities '{"showIOSLog":true}'
[Appium] Default capabilities, which will be added to each request unless overridden by desired capabilities:

[Appium]   platformName: 'iOS'

[Appium]   platformVersion: '9.3'
[Appium]   browserName: 'Safari'
[Appium]   udid: '334bde145520a3aed44d4399992a98a37461ce73'
[Appium]   fullReset: true
[Appium]   showIOSLog: true

[Appium] Appium REST http interface listener started on 127.0.0.1:4723

[HTTP] --> GET /wd/hub/status {}

[MJSONWP] Calling AppiumDriver.getStatus() with args: []

[MJSONWP] Responding to client with driver.getStatus() result: {"build":{"version":"1.5.3"...

[HTTP] <-- GET /wd/hub/status 200 25 ms - 83 

[HTTP] --> GET /wd/hub/status {}

[MJSONWP] Calling AppiumDriver.getStatus() with args: []

[MJSONWP] Responding to client with driver.getStatus() result: {"build":{"version":"1.5.3"...

[HTTP] <-- GET /wd/hub/status 200 15 ms - 83 

Launching Appium with command: '/Applications/Appium 3.app/Contents/Resources/node/bin/node' appium/build/lib/main.js --address "127.0.0.1" --debug-log-spacing --platform-version "9.3" --platform-name "iOS" --safari --browser-name "Safari" --show-ios-log --default-device

[Appium] Welcome to Appium v1.5.3

[Appium] Non-default server args:

[Appium]   address: '127.0.0.1'
[Appium]   safari: true
[Appium]   defaultDevice: true
[Appium]   debugLogSpacing: true
[Appium]   platformName: 'iOS'
[Appium]   platformVersion: '9.3'
[Appium]   browserName: 'Safari'
[Appium]   showIOSLog: true
[Appium] Deprecated server args:
[Appium]   --platform-name => --default-capabilities '{"platformName":"iOS"}'
[Appium]   --platform-version => --default-capabilities '{"platformVersion":"9.3"}'
[Appium]   --browser-name => --default-capabilities '{"browserName":"Safari"}'
[Appium]   --show-ios-log => --default-capabilities '{"showIOSLog":true}'
[Appium] Default capabilities, which will be added to each request unless overridden by desired capabilities:

[Appium]   platformName: 'iOS'
[Appium]   platformVersion: '9.3'
[Appium]   browserName: 'Safari'
[Appium]   showIOSLog: true

[Appium] Appium REST http interface listener started on 127.0.0.1:4723

[HTTP] --> GET /wd/hub/status {}

[MJSONWP] Calling AppiumDriver.getStatus() with args: []

[MJSONWP] Responding to client with driver.getStatus() result: {"build":{"version":"1.5.3"...

[HTTP] <-- GET /wd/hub/status 200 33 ms - 83 

[HTTP] --> POST /wd/hub/session {"desiredCapabilities":{"browserName":"safari","platformName":"iOS","deviceName":"iPad Mini","platformVersion":"9.3"}}

[MJSONWP] Calling AppiumDriver.createSession() with args: [{"browserName":"safari","p...

[Appium] Creating new IosDriver session
[Appium] Capabilities:
[Appium]   browserName: 'safari'
[Appium]   platformName: 'iOS'
[Appium]   deviceName: 'iPad Mini'
[Appium]   platformVersion: '9.3'
[Appium]   showIOSLog: true
[BaseDriver] Session created with session id: 6ab59511-6df9-4745-9c0c-2fc53fc60212

[debug] [iOS] Not auto-detecting udid.
[debug] [iOS] We are on iOS8+ so not copying Safari app
[debug] [iOS] Removing any remaining instruments sockets
[debug] [iOS] Cleaned up instruments socket /var/folders/z3/tkbh3cns4fd4fyd6xsdgk1bh68_4ff/T/instruments_sock
[debug] [iOS] Setting Xcode version

[debug] [iOS] Xcode version set to 7.3.1

[debug] [iOS] Setting iOS SDK Version

[debug] [iOS] iOS SDK Version set to 9.3

[debug] [iOS] Checking whether instruments supports our device string

[debug] [Instruments] Getting list of devices instruments supports

[debug] [Instruments] Instruments is at: /Applications/Xcode.app/Contents/Developer/usr/bin/instruments

[HTTP] --> GET /wd/hub/status {}

[MJSONWP] Calling AppiumDriver.getStatus() with args: []

[HTTP] <-- GET /wd/hub/status - - ms - - 

[HTTP] --> GET /wd/hub/status {}

[MJSONWP] Calling AppiumDriver.getStatus() with args: []

[HTTP] <-- GET /wd/hub/status - - ms - - 

[HTTP] --> GET /wd/hub/status {}

[MJSONWP] Calling AppiumDriver.getStatus() with args: []

[debug] [Instruments] Available devices: Apple TV 1080p (9.2) [F9576269-EAA0-4316-A9A5-44F820CE82CC] (Simulator),Apple Watch - 38mm (2.2) [9487D608-7D1C-4694-AA44-81043227D895] (Simulator),Apple Watch - 42mm (2.2) [0801BD1C-175E-4531-80B0-DBB5A80B7375] (Simulator),iPad Pro (9.3) [B52C994E-FD08-43ED-995B-D480E48D61E0] (Simulator),iPad Retina (9.3) [B69A7B96-7D78-4FB2-B254-99FF3F776FA2] (Simulator),iPhone 5 (9.2) [4425E891-D5A4-4F6C-8B9D-984F018568B9] (Simulator),iPhone 5s (9.3) [65B785AC-E473-4355-A838-6D583F9C0251] (Simulator),iPhone 6 (9.3) [C6439AFA-AF90-4551-AC75-7FCE332E1D7E] (Simulator),iPhone 6 (9.3) + Apple Watch - 38mm (2.2) [9E7BC190-87D7-4B55-AFED-7B0E1B297E68] (Simulator)

[iOSSim] Retrieving device name string for Xcode version 7.3.1

[debug] [iOSSim] Getting device string from options: {"deviceName":"iPad Mini","platformVersion":"9.3","forceIphone":false,"forceIpad":false}
[debug] [iOSSim] Final device string is 'iPad Mini (9.3)'

[iOS] Error: Could not find a device to launch. You requested 'iPad Mini (9.3)', but the available devices were: ["Apple TV 1080p (9.2) [F9576269-EAA0-4316-A9A5-44F820CE82CC] (Simulator)","Apple Watch - 38mm (2.2) [9487D608-7D1C-4694-AA44-81043227D895] (Simulator)","Apple Watch - 42mm (2.2) [0801BD1C-175E-4531-80B0-DBB5A80B7375] (Simulator)","iPad Pro (9.3) [B52C994E-FD08-43ED-995B-D480E48D61E0] (Simulator)","iPad Retina (9.3) [B69A7B96-7D78-4FB2-B254-99FF3F776FA2] (Simulator)","iPhone 5 (9.2) [4425E891-D5A4-4F6C-8B9D-984F018568B9] (Simulator)","iPhone 5s (9.3) [65B785AC-E473-4355-A838-6D583F9C0251] (Simulator)","iPhone 6 (9.3) [C6439AFA-AF90-4551-AC75-7FCE332E1D7E] (Simulator)","iPhone 6 (9.3) + Apple Watch - 38mm (2.2) [9E7BC190-87D7-4B55-AFED-7B0E1B297E68] (Simulator)"]
    at Object.wrappedLogger.errorAndThrow (lib/logger.js:60:13)
    at noDevicesError (lib/driver.js:912:14)
    at IosDriver.checkSimAvailable$ (lib/driver.js:917:9)
    at tryCatch (/Applications/Appium 3.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Applications/Appium 3.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Applications/Appium 3.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/Applications/Appium 3.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
    at run (/Applications/Appium 3.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/node_modules/core-js/library/modules/es6.promise.js:104:47)
    at /Applications/Appium 3.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/node_modules/core-js/library/modules/es6.promise.js:115:28
    at flush (/Applications/Appium 3.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/node_modules/core-js/library/modules/$.microtask.js:19:5)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
 [Error: Could not find a device to launch. You requested 'iPad Mini (9.3)', but the available devices were: ["Apple TV 1080p (9.2) [F9576269-EAA0-4316-A9A5-44F820CE82CC] (Simulator)","Apple Watch - 38mm (2.2) [9487D608-7D1C-4694-AA44-81043227D895] (Simulator)","Apple Watch - 42mm (2.2) [0801BD1C-175E-4531-80B0-DBB5A80B7375] (Simulator)","iPad Pro (9.3) [B52C994E-FD08-43ED-995B-D480E48D61E0] (Simulator)","iPad Retina (9.3) [B69A7B96-7D78-4FB2-B254-99FF3F776FA2] (Simulator)","iPhone 5 (9.2) [4425E891-D5A4-4F6C-8B9D-984F018568B9] (Simulator)","iPhone 5s (9.3) [65B785AC-E473-4355-A838-6D583F9C0251] (Simulator)","iPhone 6 (9.3) [C6439AFA-AF90-4551-AC75-7FCE332E1D7E] (Simulator)","iPhone 6 (9.3) + Apple Watch - 38mm (2.2) [9E7BC190-87D7-4B55-AFED-7B0E1B297E68] (Simulator)"]]

[HTTP] <-- POST /wd/hub/session 500 7538 ms - 925 

[MJSONWP] Encountered internal error running command: Error: Could not find a device to launch. You requested 'iPad Mini (9.3)', but the available devices were: ["Apple TV 1080p (9.2) [F9576269-EAA0-4316-A9A5-44F820CE82CC] (Simulator)","Apple Watch - 38mm (2.2) [9487D608-7D1C-4694-AA44-81043227D895] (Simulator)","Apple Watch - 42mm (2.2) [0801BD1C-175E-4531-80B0-DBB5A80B7375] (Simulator)","iPad Pro (9.3) [B52C994E-FD08-43ED-995B-D480E48D61E0] (Simulator)","iPad Retina (9.3) [B69A7B96-7D78-4FB2-B254-99FF3F776FA2] (Simulator)","iPhone 5 (9.2) [4425E891-D5A4-4F6C-8B9D-984F018568B9] (Simulator)","iPhone 5s (9.3) [65B785AC-E473-4355-A838-6D583F9C0251] (Simulator)","iPhone 6 (9.3) [C6439AFA-AF90-4551-AC75-7FCE332E1D7E] (Simulator)","iPhone 6 (9.3) + Apple Watch - 38mm (2.2) [9E7BC190-87D7-4B55-AFED-7B0E1B297E68] (Simulator)"]
    at Object.wrappedLogger.errorAndThrow (lib/logger.js:60:13)
    at noDevicesError (lib/driver.js:912:14)
    at IosDriver.checkSimAvailable$ (lib/driver.js:917:9)
    at tryCatch (/Applications/Appium 3.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:67:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Applications/Appium 3.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:315:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Applications/Appium 3.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:100:21)
    at GeneratorFunctionPrototype.invoke (/Applications/Appium 3.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/regenerator/runtime.js:136:37)
    at run (/Applications/Appium 3.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/node_modules/core-js/library/modules/es6.promise.js:104:47)
    at /Applications/Appium 3.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/node_modules/core-js/library/modules/es6.promise.js:115:28
    at flush (/Applications/Appium 3.app/Contents/Resources/node_modules/appium/node_modules/babel-runtime/node_modules/core-js/library/modules/$.microtask.js:19:5)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)

[MJSONWP] Responding to client with driver.getStatus() result: {"build":{"version":"1.5.3"...

[MJSONWP] Responding to client with driver.getStatus() result: {"build":{"version":"1.5.3"...
[MJSONWP] Responding to client with driver.getStatus() result: {"build":{"version":"1.5.3"...

[HTTP] <-- GET /wd/hub/status 200 418 ms - 83 



Please help me up

**Comment by imurchie** [comment]
This looks like Appium is being launched twice. Once with a udid in the server arguments and once without. And then you're getting a simulator test, but don't have a simulator for your device.

I would suggest adding the udid to your desired capabilities.
